src/engine/Drawing/Animation.ts

```diff
@@ -277,24 +280,35 @@ export class AnimationImpl implements Drawable {
   public draw(options: DrawOptions): void;
   public draw(ctxOrOptions: CanvasRenderingContext2D | DrawOptions, x?: number, y?: number) {
     if (ctxOrOptions instanceof CanvasRenderingContext2D) {
-      this._drawWithOptions({ ctx: ctxOrOptions, x, y, flipHorizontal: this.flipHorizontal, flipVertical: this.flipVertical });
+      this._drawWithOptions({ ctx: ctxOrOptions, x, y });
     } else {
       this._drawWithOptions(ctxOrOptions);
     }
   }
 
   private _drawWithOptions(options: DrawOptions) {
+    const animOptions = {
+      ...options,
+      rotation: nullish(options.rotation, this.rotation),
```

Comment:
Does nullish account for 0 number or empty strings, since those aren't nullish? Sorry, I can't see the impl easily on my phone